packages/react-aria-components/src/GridList.tsx

```diff
@@ -599,33 +600,35 @@ export const GridListSection = /*#__PURE__*/ createBranchComponent(SectionNode,
   delete DOMProps.id;
 
   return (
-    <section
+    <div
```

Comment:
see Michael's comment for context regarding this change, but as a result, i needed to update the ref types. but that caused issues with the HeaderContext since it expects an HTMLElement. so i had to create a new context for GridListHeader rather than reuse the HeaderContext